app/models/host.rb

```diff
@@ -88,11 +88,17 @@ class Host < ApplicationRecord
     left_outer_joins(:test_result_profiles)
   }
 
-  scope :with_failed_rules_count, lambda {
-    sq = Host.left_outer_joins(test_result_profiles: :rule_results)
-             .where(rule_results: { result: RuleResult::FAILED })
+  scope :with_failed_rules_count, lambda { |profile = nil|
+    profile ||= RequestStore.store['scoped_search_context_profiles']
+    profile_ids = profile&.pluck(:id) || []
+
+    sq = Host.left_outer_joins(test_results: :rule_results)
+             .merge(TestResult.latest('LEFT OUTER JOIN'))
```

Comment:
Scoping down the results to the latest one, but also taking into account of systems without test results (that's why the `LEFT OUTER JOIN`)

Reply:
This was a good catch 👍 I forgot there could be older test results in the database